cwltool/singularity.py

```diff
@@ -280,16 +280,17 @@ def get_from_requirements(
 
     @staticmethod
     def append_volume(
-        runtime: List[str], source: str, target: str, writable: bool = False
+            runtime: List[str], source: str, target: str, writable: bool = False
     ) -> None:
-        runtime.append("--bind")
-        runtime.append(
-            "{}:{}:{}".format(
-                docker_windows_path_adjust(source),
-                docker_windows_path_adjust(target),
-                "rw" if writable else "ro",
-            )
-        )
+        src = docker_windows_path_adjust(source)
+        dst = docker_windows_path_adjust(target)
+        writable = "rw" if writable else "ro"
+        if os.path.isfile(src) and dst.endswith(os.path.basename(src)) and writable == 'ro':
+            src = os.path.dirname(src)
+            dst = os.path.dirname(dst)
+        bind_arg = f"--bind={src}:{dst}:{writable}"
+        if bind_arg not in runtime:
```

Comment:
I have tried running this PR after getting the `argument list too long` error, and it seems that the `source` directory here is actually different for each file, and therefore it did not reduce the number of `--bind` arguments:

```
--bind=/juno/work/access/testing/users/johnsoni/access_qc/5500_HF/jobStore/files/for-job/kind-CWLJob/instance-aop8iz_w/file-34cfcc9ed208473bb67710a5108f1a32:/var/lib/cwl/s5g5ff703f1-41b2-56d3-93b3-7e36c1a37224/all_qc_files/duplex_bam_sequence_qc_dir/C-R0CLHU-N001-d:ro \
--bind=/juno/work/access/testing/users/johnsoni/access_qc/5500_HF/jobStore/files/for-job/kind-CWLJob/instance-aop8iz_w/file-fee328e946c9448c92ee501c6daa4b5a:/var/lib/cwl/s5g5ff703f1-41b2-56d3-93b3-7e36c1a37224/all_qc_files/duplex_bam_sequence_qc_dir/C-R0CLHU-N001-d:ro \
```

It seems toil actually provides different parent directories to files that might be part of the same Directory object